app/src/main/java/edu/mills/cs115/fruitthief/MainActivity.kt

```diff
@@ -0,0 +1,118 @@
+import edu.mills.cs115.fruitthief.database.PopulateFruitTable
+import edu.mills.cs115.fruitthief.database.PopulateTreeTable
+import edu.mills.cs115.fruitthief.map.MapFragment
+import kotlinx.coroutines.runBlocking
+
+class MainActivity : AppCompatActivity() { //, OnMapReadyCallback {
+
+    private lateinit var binding: ActivityMainBinding
+    private lateinit var navController: NavController
+    private lateinit var drawerLayout: DrawerLayout
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
+        navController = findNavController(R.id.nav_host_fragment)
+        drawerLayout = binding.drawerLayout
+        val application = requireNotNull(this).application
+        val dataSource = FruitTreeDatabase.getInstance(application).fruitTreeDAO
+
+        // clear tables for testing
+        runBlocking {
+            // clear for testing
+            dataSource.clearFruitTable()
+            dataSource.clearTrees()
+            PopulateFruitTable(dataSource)
+            PopulateTreeTable(dataSource)
+        }
+
+        // GPS Permission
+        if (ContextCompat.checkSelfPermission(this@MainActivity,
+                Manifest.permission.ACCESS_FINE_LOCATION) !==
+            PackageManager.PERMISSION_GRANTED) {
+            if (ActivityCompat.shouldShowRequestPermissionRationale(this@MainActivity,
+                    Manifest.permission.ACCESS_FINE_LOCATION)) {
+                ActivityCompat.requestPermissions(this@MainActivity,
+                    arrayOf(Manifest.permission.ACCESS_FINE_LOCATION), 1)
+            } else {
+                ActivityCompat.requestPermissions(this@MainActivity,
+                    arrayOf(Manifest.permission.ACCESS_FINE_LOCATION), 1)
+            }
+        }
+
+        // Map
+        if (savedInstanceState == null) {
+            supportFragmentManager.commit {
+                setReorderingAllowed(true)
+                add<MapFragment>(R.id.fragment_container_view)
+            }
+        }
+
+        setUpNavigation()
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        menuInflater.inflate(R.menu.main_menu_drawer, menu)
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return NavigationUI.
+        onNavDestinationSelected(item, navController)
+                || super.onOptionsItemSelected(item)
+    }
+
+    override fun onSupportNavigateUp(): Boolean {
+        val navController = findNavController(R.id.nav_host_fragment)
+        return NavigationUI.navigateUp(navController, drawerLayout)
+    }
+
+    private fun setUpNavigation() {
+        NavigationUI.setupWithNavController(binding.navView, navController)
+        val toolbar: Toolbar = findViewById(R.id.toolbar)
```

Comment:
Can you use `binding.toolbar` rather than calling `findViewById()`? https://inthecheesefactory.com/blog/say-goodbye-to-findviewbyid-with-data-binding-library/en